plugins/module_utils/cloud_py.py

```diff
@@ -0,0 +1,185 @@
+            backoff (int or float): backoff multiplier e.g. value of 2 will  double the delay each retry
+                default=2
+            max_delay (int or None): maximum amount of time to wait between retries.
+                default=60
+            catch_extra_error_codes: Additional error messages to catch, in addition to those which may be defined by a subclass of CloudRetry
+                default=None
+        Returns:
+            Callable: A generator that calls the decorated function using an exponential backoff.
+        """
+        def exponential_backoff_decorator(func):
+            @functools.wraps(func)
+            def exponential_retry_wrapper(*args, **kwargs):
+                partial_func = functools.partial(func, *args, **kwargs)
+                sleep_time_generator = _backoff_value_generator(delay=delay, backoff=backoff, max_delay=max_delay)
+                return _retry_func(
+                    func=partial_func,
+                    sleep_time_generator=sleep_time_generator,
+                    retries=retries,
+                    catch_extra_error_codes=catch_extra_error_codes,
+                    found_f=cls.found,
+                    status_code_from_except_f=cls.status_code_from_exception
+                )
+            return exponential_retry_wrapper
+
+        return exponential_backoff_decorator
+
+    @classmethod
+    def jittered_backoff(cls, retries=10, delay=3, max_delay=60, catch_extra_error_codes=None):
+        """Wrap a callable with retry behavior.
+        Args:
+            retries (int): Number of times to retry a failed request before giving up
+                default=10
+            delay (int or float): Initial delay between retries in seconds
+                default=3
+            max_delay (int or None): maximum amount of time to wait between retries.
+                default=60
+            catch_extra_error_codes: Additional error messages to catch, in addition to those which may be defined by a subclass of CloudRetry
+                default=None
+        Returns:
+            Callable: A generator that calls the decorated function using using a jittered backoff strategy.
+        """
+        def jitter_backoff_decorator(func):
+            @functools.wraps(func)
+            def jitter_retry_wrapper(*args, **kwargs):
+                partial_func = functools.partial(func, *args, **kwargs)
+                sleep_time_generator = _backoff_value_generator(delay=delay, backoff=2.0, max_delay=max_delay)
+                return _retry_func(
+                    func=partial_func,
+                    sleep_time_generator=sleep_time_generator,
+                    retries=retries,
+                    catch_extra_error_codes=catch_extra_error_codes,
+                    found_f=cls.found,
+                    status_code_from_except_f=cls.status_code_from_exception
+                )
+            return jitter_retry_wrapper
+
+        return jitter_backoff_decorator
```

Comment:
this magic feels like it's something that should be refactored into a base_decorator which may make things easier if someone needed to implement their own backoff.